pkg/registries/registries_test.go

```diff
@@ -76,155 +77,229 @@ func TestIsValidRegistriesConfScope(t *testing.T) {
-					// Opposite order of the vendor’s mirrors.
-					// WARNING: The order in this case is unspecified by the ICSP specification, and may change at any time;
-					// this test case only ensures that the corner case is handled reasonably, and that the output is stable
-					// (i.e. the operator does not cause unnecessary changes in output objects.)
-					{Source: "source.vendor2.com", Mirrors: []string{"local-mirror2.example.com", "registry2.vendor2.com", "registry1.vendor2.com"}},
-				},
+		result: []mergedMirrorSet{
+			{source: "source.example.com", mirrors: []string{"z1.example.com", "source.example.com", "y2.example.com", "x3.example.com"}},
+		},
+	},
+	{
+		// Worst case of the above: _only_ the source included in mirrors, even perhaps several times.
+		name: "Mirrors includes only source",
+		input: [][]mergedMirrorSet{
+			{
+				{source: "source.example.com", mirrors: []string{"source.example.com"}},
+				{source: "source.example.net", mirrors: []string{"source.example.net", "source.example.net", "source.example.net"}},
+			},
+		},
+		result: []mergedMirrorSet{},
+	},
+	// More complex mirror set combinations are mostly tested in TestTopoGraph
+	{
+		name: "Example",
+		input: [][]mergedMirrorSet{
+			{ // Vendor-provided default configuration
+				{source: "source.vendor.com", mirrors: []string{"registry2.vendor.com"}},
+			},
+			{ // Vendor2-provided default configuration
+				{source: "source.vendor2.com", mirrors: []string{"registry1.vendor2.com", "registry2.vendor2.com"}},
 			},
-			result: []apioperatorsv1alpha1.RepositoryDigestMirrors{
-				{Source: "source.vendor.com", Mirrors: []string{"local-mirror.example.com", "registry2.vendor.com"}},
-				{Source: "source.vendor2.com", Mirrors: []string{"local-mirror2.example.com", "registry1.vendor2.com", "registry2.vendor2.com"}},
+			{ // Admin-configured local mirrors:
+				{source: "source.vendor.com", mirrors: []string{"local-mirror.example.com"}},
+				// Opposite order of the vendor’s mirrors.
+				// WARNING: The order in this case is unspecified by the ICSP specification, and may change at any time;
+				// this test case only ensures that the corner case is handled reasonably, and that the output is stable
+				// (i.e. the operator does not cause unnecessary changes in output objects.)
+				{source: "source.vendor2.com", mirrors: []string{"local-mirror2.example.com", "registry2.vendor2.com", "registry1.vendor2.com"}},
 			},
 		},
-	} {
-		t.Run(c.name, func(t *testing.T) {
+		result: []mergedMirrorSet{
+			{source: "source.vendor.com", mirrors: []string{"local-mirror.example.com", "registry2.vendor.com"}},
+			{source: "source.vendor2.com", mirrors: []string{"local-mirror2.example.com", "registry1.vendor2.com", "registry2.vendor2.com"}},
+		},
+	},
+}
+
+func TestMergedICSPMirrorSets(t *testing.T) {
+	for _, tc := range mergedMirrorsetsTestcases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.name == "Separate mirror sets with mirrorSourcePolicy set" {
```

Comment:
Please make this driven by a real data field.

Either here,
```go
if tc.requiresMirrorSourcePolicy { t.Skip() }
```
(and set the `tc.requiresMirrorSourcePolicy` field on the test case), or (preferably?) make this fully automatic below, with something like

```
if item.mirrorSourcePolicy != "" {
    t.Skip("…")
}
```